daily_read/order_portal.py

```diff
@@ -42,69 +44,69 @@ def get_orders(self, node=None, status=None, orderer=None, recent=True):
-        else:
-            response = self.__get(f"/api/v1/account/{orderer}/orders", params)
+            params["year"] = "all"
+        if orderer:
+            params["owner"] = orderer
+
+        response = self._get("api/v1/orders", params)
 
         try:
-            self.all_orders = response.json()["orders"]
+            self.all_orders = self.all_orders + response.json()["items"]
         except requests.exceptions.JSONDecodeError as e:
             log.critical(
                 f"Could not fetch orders for {{node: {node}, status: {status}, orderer={orderer}, recent={recent}}}"
             )
             raise
-        log.info(f"Fetched {len(self.all_orders)} order(s) from the Order Portal")
+        log.info(f"Fetched a total of {len(self.all_orders)} order(s) from the Order Portal")
 
-    def process_orders(self, use_node="", closed_before_in_days=7):
-        """process orders"""
+    def process_orders(self, closed_before_in_days=30):
+        """Process orderers orders to select ones that need to be updated"""
 
         order_updates = {}
         pull_date = datetime.datetime.now()
         older_than_cutoff = (pull_date - datetime.timedelta(days=closed_before_in_days)).date()
-        statusdb_sess = StatusdbSession(self.config_values, db="projects")
         for order in self.all_orders:
-            process_order = True
-            if use_node:
-                process_order = order.get("fields", {}).get("assigned_node", "") == use_node
-
-            if process_order:
-                if (
-                    order["status"] == "closed"
-                    and datetime.datetime.strptime(order["history"]["closed"], "%Y-%m-%d").date() >= older_than_cutoff
-                ):
-                    continue
-                proj_info = statusdb_sess.get_entry(order["identifier"])
-                if proj_info:
-                    proj_info[order["identifier"]]["pull_date"] = f"{pull_date}"
-                    proj_info[order["identifier"]]["iuid"] = order["iuid"]
-                    pi_email = order["owner"]["email"]
-                    if pi_email in order_updates:
-                        order_updates[pi_email]["projects"].append(proj_info[order["identifier"]])
-                    else:
-                        order_updates[pi_email] = {
-                            "pull_date": f"{pull_date}",
-                            "projects": [proj_info[order["identifier"]]],
-                        }
+            if (
+                order["status"] == "closed"
+                and datetime.datetime.strptime(order["history"]["closed"], "%Y-%m-%d").date() >= older_than_cutoff
+            ):
+                continue
+            if order["identifier"] in self.projects_data.data.keys():
+                proj_info = self.projects_data.data[order["identifier"]]
+                if order["reports"]:
+                    proj_prog_rep = next(item for item in order["reports"] if item["name"] == "Project Progress")
```

Comment:
Not sure I follow the logic with the `next`. It returns one item from the list of project progress reports? Yeah, that might make sense. Perhaps add a comment?

Reply:
I think this also will fail if there are no "Project Progress" report but there are other reports (rare occassion but still possible)

Reply:
Issue submitted:
https://github.com/NationalGenomicsInfrastructure/DailyRead/issues/18